include/dos_system.h

```diff
@@ -414,10 +425,14 @@ typedef bool (MultiplexHandler)(void);
 void DOS_AddMultiplexHandler(MultiplexHandler* handler);
 void DOS_DeleteMultiplexHandler(MultiplexHandler* handler);
 
-/* AddDevice stores the pointer to a created device */
-void DOS_AddDevice(DOS_Device * adddev);
-/* DelDevice destroys the device that is pointed to. */
-void DOS_DelDevice(DOS_Device * dev);
+// Add (or remove) the device in the DOS device registry. Managed devices are
+// externally owned and managed per RAII. Unmanaged devices will be destroyed by
+// the device registry on close. Note that when adding, duplicate-named devices
+// are removed from the registry prior to adding because devices must be
+// uniquely named.
+void DOS_AddUnmanagedDevice(DOS_Device* unmanaged_dev);
+void DOS_AddManagedDevice(DOS_Device* managed_dev);
```

Comment:
🎖️ for not just doing ` void DOS_AddDevice(DOS_Device* dev, const bool is_managed);`

Reply:
```
DOS_AddDevice(com1, true);
DOS_AddDevice(ems, false);
DOS_AddDevice(nul, true);
```

Just lookup the API, bro :trollface: